synapse/handlers/cas.py

```diff
@@ -120,7 +121,10 @@ async def _validate_ticket(
         Returns:
             The parsed CAS response.
         """
-        uri = self._cas_server_url + "/proxyValidate"
+        if self._cas_protocol_version == 3:
+            uri = self._cas_server_url + "/p3/proxyValidate"
```

Comment:
Seems legit -> spec docs for CAS protocol version 3: https://apereo.github.io/cas/6.6.x/protocol/CAS-Protocol-Specification.html#29-p3proxyvalidate-cas-30

> 2.9. /p3/proxyValidate [CAS 3.0]
>
> `/p3/proxyValidate` MUST perform the same validation tasks as `/p3/serviceValidate` and additionally validate proxy tickets. See Section [2.8](https://apereo.github.io/cas/6.6.x/protocol/CAS-Protocol-Specification.html#head2.5).

> 2.8. /p3/serviceValidate [CAS 3.0]
> 
> `/p3/serviceValidate` MUST perform the same validation tasks as /serviceValidate and additionally return user attributes in the CAS response. See Section [2.5](https://apereo.github.io/cas/6.6.x/protocol/CAS-Protocol-Specification.html#head2.5) and Section [2.5.7](https://apereo.github.io/cas/6.6.x/protocol/CAS-Protocol-Specification.html#head2.5.7) for details.

> 2.5.7 Example response with custom attributes
>
> ```xml
> <cas:serviceResponse xmlns:cas="http://www.yale.edu/tp/cas">
>   <cas:authenticationSuccess>
>     <cas:user>username</cas:user>
>     <cas:attributes>
>       <cas:firstname>John</cas:firstname>
>       <cas:lastname>Doe</cas:lastname>
>       <cas:title>Mr.</cas:title>
>       <cas:email>jdoe@example.org</cas:email>
>       <cas:affiliation>staff</cas:affiliation>
>       <cas:affiliation>faculty</cas:affiliation>
>     </cas:attributes>
>     <cas:proxyGrantingTicket>PGTIOU-84678-8a9d...</cas:proxyGrantingTicket>
>   </cas:authenticationSuccess>
> </cas:serviceResponse>
> ```